src/test/java/teammates/test/cases/webapi/GetInstructorPrivilegeActionTest.java

```diff
@@ -20,16 +31,236 @@ protected String getRequestMethod() {
         return GET;
     }
 
+    @Override
+    protected void prepareTestData() {
+        InstructorAttributes instructor1ofCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        String section1 = dataBundle.students.get("student1InCourse1").getSection();
+        String session1 = dataBundle.feedbackSessions.get("session1InCourse1").getFeedbackSessionName();
+        InstructorPrivileges privileges = instructor1ofCourse1.privileges;
+        // update section privilege for testing purpose.
+
+        // course level privilege
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS, false);
+
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS, false);
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS, false);
+        privileges.updatePrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS, false);
+
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE, true);
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT, true);
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION, false);
+        privileges.updatePrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR, false);
+
+        // section level privilege
+        privileges.updatePrivilege(section1,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS, true);
+
+        // session level privilege
+        privileges.updatePrivilege(section1, session1,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS, true);
+        privileges.updatePrivilege(section1, session1,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS, true);
+
+        instructor1ofCourse1.privileges = privileges;
+
+        instructor1ofCourse1.role = Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_CUSTOM;
+        dataBundle.instructors.put("instructor1OfCourse1", instructor1ofCourse1);
+        removeAndRestoreDataBundle(dataBundle);
+    }
+
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        InstructorAttributes instructor1ofCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        FeedbackSessionAttributes session1ofCourse1 = dataBundle.feedbackSessions.get("session1InCourse1");
+        StudentAttributes student1InCourse1 = dataBundle.students.get("student1InCourse1");
+        loginAsInstructor(instructor1ofCourse1.getGoogleId());
+
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+
+        ______TS("Request with course id to fetch general privileges");
```

Comment:
Another suggestion for you which is also a good practice.

Instead of using `______TS()`, you can separate them in to separated `@Test` method and follow the test naming convention.